services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/ProtectedPathSpec.groovy

```diff
@@ -635,11 +643,29 @@ class ProtectedPathSpec extends BaseSpecification {
     }
 
     def "Able to swap main and protected paths manually"() {
-        given: "Two active not neighboring switches with two possible paths at least"
-        def (Switch srcSwitch, Switch dstSwitch) = getNotNeighboringSwitchPair(2)
+        given: "A simple flow"
+        def switches = topology.activeTraffGens*.switchConnected
+        def allLinks = northbound.getAllLinks()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = database.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            /**just for sure that we have non overlapping path available
+             * and we make assumption that parallel links won't be used by a flow
+             * because another switch is available */
+            List<List<Switch>> pathSwitches = possibleFlowPaths.collect { pathHelper.getInvolvedSwitches(it) }
+            Integer numberOfDifferentSwitchInPath = 0
+            for (int i = 0; i < pathSwitches.size() - 1; i++) {
+                if (!pathSwitches[i][1..-2].containsAll(pathSwitches[i + 1][1..-2])) {
+                    numberOfDifferentSwitchInPath += 1
+                }
+            }
+            allLinks.every { link ->
+                !(link.source.switchId == src.dpId && link.destination.switchId == dst.dpId)
+            } && possibleFlowPaths.size() >= 2 && numberOfDifferentSwitchInPath >= 2
+        } ?: assumeTrue("No suiting switches found", false)
```

Comment:
I find the above code close to not maintainable. You are using old ways to obtain switches, which is being deprecated and refactored out. You can achieve the same using new methods in a much more readable and laconic way
```
def tgSwitches = topologyDefinition.getActiveTraffGens()*.getConnectedSwitch()
def (Switch srcSwitch, Switch dstSwitch) = topologyHelper.getAllNotNeighboringSwitchPairs().findAll { 
  def allowsTraffexam = it.src in tgSwitches && it.dst in tgSwitches 
  def usesUniqueSwitches = it.paths.collectMany { pathHelper.getInvolvedSwitches(it) }
                   .unique { it.dpId }.size() > 3
  return allowsTraffexam && usesUniqueSwitches
}
```

Reply:
done